src/commonTest/kotlin/kotools/types/collection/NotEmptyMapTest.kt

```diff
@@ -75,38 +165,18 @@ class NotEmptyMapTest {
         )
     }
 
-    @Test
-    fun notEmptyMapOf_should_pass() {
-        val head: Pair<String, Int> = "a" to Random.nextInt()
-        val tail: Array<Pair<String, Int>> = arrayOf(
-            "b" to Random.nextInt(),
-            "c" to Random.nextInt()
-        )
-        assertEquals(
-            actual = notEmptyMapOf(head, *tail).toMap(),
-            expected = mapOf(head, *tail)
-        )
-    }
+}
 
+class NotEmptyMapIntegrationTest {
     @Test
-    fun map_toNotEmptyMap_should_pass_with_a_not_empty_Map() {
-        val map: Map<String, Int> = mapOf(
-            "a" to Random.nextInt(),
-            "b" to Random.nextInt(),
-            "c" to Random.nextInt()
-        )
-        val result: Result<NotEmptyMap<String, Int>> = map.toNotEmptyMap()
-        result.getOrThrow()
-            .toMap()
-            .entries contentShouldEqual map.entries
-    }
+    fun updating_the_original_MutableMap_should_not_impact_the_NotEmptyMap() {
+        val original: MutableMap<Char, Int> = mutableMapOf('a' to 1, 'b' to 2)
+        val notEmptyMap: NotEmptyMap<Char, Int> = original.toNotEmptyMap()
+            .getOrThrow()
+        assertEquals("$original", "$notEmptyMap")
 
-    @Test
-    fun map_toNotEmptyMap_should_fail_with_an_empty_Map() {
-        val result: Result<NotEmptyMap<String, Int>> = emptyMap<String, Int>()
-            .toNotEmptyMap()
-        assertFailsWith<IllegalArgumentException>(block = result::getOrThrow)
-            .shouldHaveAMessage()
+        original.clear()
+        assertNotEquals("$original", "$notEmptyMap")
```

Comment:
Use the assertions available in the [`kotools.types.Assertions.kt`](https://github.com/kotools/types/blob/5d1dc8f98f2220e96081381809cde9d261d04f8f/src/commonTest/kotlin/kotools/types/Assertions.kt) file.

```suggestion
        "$notEmptyMap" shouldNotEqual "$original"
```